CONTRIBUTING.md

```diff
@@ -50,54 +50,56 @@ Codespaces is new technology that allows you to use a container as your developm
+#### -> CMake
 
 CMake version 3.13 or higher is required to build these libraries. Download and install CMake from the project's
 [website](https://cmake.org/download/).
 
-##### Third Party Dependencies
+### Third Party Dependencies
 
-- curl
-- libxml2
-- clang-format (min version 10)
+Azure SDK use Vcpkg manifest mode to declare the [list of required dependencies](https://github.com/Azure/azure-sdk-for-cpp/blob/main/vcpkg.json) for building the SDK service libraries. It will also get and set up Vcpkg automatically. You can move on to [Building the project](#building-the-project) and skip next part if you are not interested in learning the alternatives for setting up dependencies.
 
-Vcpkg can be used to install the Azure SDK for CPP dependencies into a specific folder on the system instead of globally installing them.
-Follow [vcpkg install guide](https://github.com/microsoft/vcpkg#getting-started) to get vcpkg and install the following dependencies:
+If you didn't jump to `Build the project`, you will now learn about how you can control Vcpkg and dependencies integration.
 
-```sh
-./vcpkg install curl libxml2
-```
+#### Setting up Vcpkg automatic integration
 
-When using vcpkg, you can set the `VCPKG_ROOT` environment variable to the vcpkg Git repository folder. This would automatically set the CMake variable `CMAKE_TOOLCHAIN_FILE` for you, enabling the project to use any library installed with vcpkg.
+If the CMake option _-DCMAKE_TOOLCHAIN_FILE=..._ is not defined to generate the project, the Azure SDK project will automatically get Vcpkg and link it to get its dependencies. You can use the next environment variables to change this behavior:
 
-The Azure SDK for C++ uses [this vcpkg release version](https://github.com/Azure/azure-sdk-for-cpp/blob/main/eng/vcpkg-commit.txt) for continuous integration (CI) building and testing. Make sure to checkout this version when following the next steps for building and running the Azure SDK for C++. Using a newer vcpkg version might still work, however, if it is tested.
-
-```sh
-# Checking out vcpkg release version before installing dependencies
+<center>
 
-git clone https://github.com/Microsoft/vcpkg.git
-cd vcpkg
-# Checkout the vcpkg commit from the vcpkg-commit.txt file (link above)
-git checkout <vcpkg commit>
+<table>
+<tr>
+<td>Environment Variable</td>
+<td>Description</td>
+</tr>
+<tr>
+<td>AZURE_SDK_DISABLE_AUTO_VCPKG</td>
+<td>When defined, Vcpkg won't be automatically cloned and linked. Use this setting, for example, if your dependencies are installed in the system and you don't need to get them.</td>
+</tr>
+<tr>
+<td>AZURE_SDK_VCPKG_COMMIT</td>
+<td>This variable can set the git commit id to be used when automatically cloning Vcpkg.</td>
+</tr>
+<tr>
+<td>VCPKG_ROOT</td>
+<td>Use this variable to set an existing Vcpkg folder from your system to be linked for building. Use this, for example, when working with Vcpkg classic mode, to switch between different Vcpkg folders.</td>
```

Comment:
It's interesting to use "system" to refer to the developer machine or computer. I suspect most folks are using a desktop machine (whether local or remote) to contribute to the repo, so "machine" or "computer" felt more relevant to the reader.

That said, I suppose, if we include VMs, codespaces, and other such environments, a general term like system could work better.